src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -55,31 +52,42 @@
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
     public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
 
-    private final Index index;
     private final EditPersonDescriptor editPersonDescriptor;
+    private final Person personToEdit;
 
     /**
-     * @param index                of the person in the filtered person list to edit
      * @param editPersonDescriptor details to edit the person with
      */
-    public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
-        requireNonNull(index);
+    public EditCommand(EditPersonDescriptor editPersonDescriptor) {
         requireNonNull(editPersonDescriptor);
+        personToEdit = Person.getEmptyPerson();
+        this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+    }
 
-        this.index = index;
+    private EditCommand(EditPersonDescriptor editPersonDescriptor, Person personToEdit) {
+        requireNonNull(editPersonDescriptor);
+        requireNonNull(personToEdit);
+        this.personToEdit = new Person(personToEdit);
```

Comment:
Why do we need to create a copy of the original person?

Reply:
I think I did that because of immutability since there wass also one for the initializing of the following this.editPersonDescriptor

Reply:
Could be just a case of me inferring wrongly so correct me if I am wrong